src/api/article/controller.js

```diff
@@ -11,74 +11,44 @@ import {
 } from 'api/helpers/metadata';
 
 import { User } from 'api/user';
-import Article, {
-  serializeArticle,
-  checkIsPublished,
-  queryUnpublished,
-  POPULATE_OPTIONS,
-} from './article.model';
+import Article, { checkIsPublished, DEFAULT_ARTICLE_QUERY } from './article.model';
 import ArticleBrand from './brand/model';
 import ArticleCollection from './collection/model';
 import LocalizedArticle from './localized/model';
 
-export const getAll = ({ query, user }, res, next) => {
-  const skip = parseInt(query.skip) || 0; // eslint-disable-line radix
-  // A limit() value of 0 is equivalent to setting no limit.
-  const take = parseInt(query.take) || 0; // eslint-disable-line radix
-  let data;
-  const articlesQuery = {
-    $and: [
-      {
-        active: true,
-        // TODO(uladbohdan): investigate how it was saved to db
-        // https://trello.com/c/uLai119m/265-empty-locales-slug-duplication-bug
-        locales: { $exists: true },
-      },
-      queryUnpublished(user),
-    ],
-  };
-
-  return Article.find(articlesQuery)
-    .populate('author', POPULATE_OPTIONS.author)
-    .populate('brand', POPULATE_OPTIONS.brand)
-    .populate(POPULATE_OPTIONS.collection(user))
-    .populate(POPULATE_OPTIONS.locales)
-    .populate(POPULATE_OPTIONS.metadata)
-    .sort({ publishAt: 'desc' })
-    .skip(skip)
-    .limit(take)
-    .then(articles => articles.map(serializeArticle))
-    .then(articles => {
-      data = articles;
-      return Article.find(articlesQuery).countDocuments();
-    })
-    .then(count => ({
+export const getAll = ({ query: { skip, take }, user }, res, next) =>
+  Article.query({
+    query: DEFAULT_ARTICLE_QUERY(user),
+    user,
+    sort: { publishAt: 'desc' },
+    skip: parseInt(skip) || 0, // eslint-disable-line radix
+    // A limit() value of 0 is equivalent to setting no limit.
+    limit: parseInt(take) || 0, // eslint-disable-line radix
+  })
+    .then(async data => ({
       data,
-      total: count,
+      total: await Article.find(DEFAULT_ARTICLE_QUERY(user)).countDocuments(),
     }))
     .then(sendJson(res))
     .catch(next);
-};
 
 const retrieveArticleId = (slugOrId, options) =>
   LocalizedArticle.findOne({ slug: slugOrId, ...options })
     .then(result => (result && result.articleId) || (isValidId(slugOrId) && slugOrId))
     .then(checkIsFound);
 
 const getArticleById = (articleId, user) =>
-  Article.findOne({ _id: articleId, active: true })
-    .populate('author', POPULATE_OPTIONS.author)
-    .populate('brand', POPULATE_OPTIONS.brand)
-    .populate(POPULATE_OPTIONS.collection(user))
-    .populate(POPULATE_OPTIONS.locales)
-    .populate(POPULATE_OPTIONS.metadata);
+  Article.query({
+    query: { _id: articleId, active: true },
+    user,
+    limit: 1,
+  }).then(articles => articles[0]);
```

Comment:
is the `find({ limit: 1 })` the same by performance as `findOne()`?

Reply:
https://stackoverflow.com/questions/33156703/whats-faster-find-limit1-or-findone-in-mongodb-mongoose/33156959